index.html

```diff
@@ -410,6 +463,94 @@ <h3 id="installation-sec">
       </h2>
       <section>
         <h3>
+          <code>BeforeInstallPromptEvent</code>
+        </h3>
+        <pre class="idl">
+enum AppBannerPromptOutcome {
+  "accepted",
+  "dismissed"
+};
+
+[Constructor(DOMString typeArg, optional BeforeInstallPromptEventInit eventInit)]
+interface BeforeInstallPromptEvent : Event {
+    readonly attribute Promise&lt;AppBannerPromptOutcome&gt; userChoice;
+    Promise&lt;void&gt; prompt();
+};
+
+dictionary BeforeInstallPromptEventInit : EventInit {
+  AppBannerPromptOutcome userChoice;
```

Comment:
This is something you've added here (that isn't in Chrome), right?

TBH it's not clear to me why we want to allow web developers to construct their own `BeforeInstallPromptEvent` objects that aren't being delivered by the UA. There isn't much use to this. If we do want to allow it for completeness, we need to make sure that it's not possible to use these fabricated BIPE objects to show a prompt -- more on that later.

It's strange that we go to the effort of allowing the website to specify a `userChoice` here (to simulate a real BIPE) but we don't provide any mechanism for them to override the `prompt` method. I suppose they could just subclass `BeforeInstallPromptEvent`, but if we allow that, then they can also override `userChoice` in the subclass. So I still don't see the point of this.

But if we must let the client supply a userChoice here, why is it an `AppBannerPromptOutcome` and not a `Promise<AppBannerPromptOutcome>`? The way it is now, the client has to pass the user's actual choice (accepted or dismissed) to the constructor of the event, before the simulated prompt is even shown, and the promise gets immediately resolved during construction. Shouldn't this constructor take a promise that can be resolved at a later time?